bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java

```diff
@@ -109,6 +118,38 @@ public void reduceNumberOfDiscoveryServices() {
 
     private final Logger logger = LoggerFactory.getLogger(DiscoveryServiceRegistryImpl.class);
 
+    private Inbox inbox;
+
+    private ThingRegistry thingRegistry;
+
+    private DiscoveryServiceCallback discoveryServiceCallback = new DiscoveryServiceCallback() {
+
+        @Override
+        public Thing getExistingThing(ThingUID thingUID) {
+            if (thingRegistry == null) {
```

Comment:
Is this check save?
The OSGi framework calls to set / unset thing registry are async.
Don't you need something similar to this one:
```java
ThingRegistry tmp = thingRegistry;
if (tmp == null) {
    ...
} else {
    return tmp.get(thingUID);
}